includes/class-fs-plugin-updater.php

```diff
@@ -120,6 +120,75 @@ private function filters() {
             }
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.1.4
+         */
+        function catch_plugin_information_dialog_contents() {
+            if (
+                'plugin-information' !== fs_request_get( 'tab', false ) ||
+                $this->_fs->get_slug() !== fs_request_get( 'plugin', false )
+            ) {
+                return;
+            }
+
+            add_action( 'admin_footer', array( &$this, 'edit_and_echo_plugin_information_dialog_contents' ), 0, 1 );
+
+            ob_start();
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.1.4
+         *
+         * @param string $hook_suffix
+         */
+        function edit_and_echo_plugin_information_dialog_contents( $hook_suffix ) {
+            if (
+                'plugin-information' !== fs_request_get( 'tab', false ) ||
+                $this->_fs->get_slug() !== fs_request_get( 'plugin', false )
+            ) {
+                return;
+            }
+
+            $license = $this->_fs->_get_license();
+
+            $subscription = ( is_object( $license ) && ! $license->is_lifetime() ) ?
+                $this->_fs->_get_subscription( $license->id ) :
+                null;
+
+            $contents = ob_get_clean();
+
+            /**
+             * Replace the plugin information dialog's "Install Update Now" button's text and URL. If there's a license,

```

Comment:
@fajardoleo thanks for the explanation. Question -> why do we "hack" it instead of modifying it in the `fs-plugin-info-dialog.php`? As far as I know, we have a full control over the rendering of the plugin's info dialog.

Reply:
@vovafeldman, `fs-plugin-info-dialog.php` is used only for add-ons.